masonry_core/src/core/widget_pod.rs

```diff
@@ -13,89 +15,120 @@ pub struct WidgetPod<W: ?Sized> {
     inner: WidgetPodInner<W>,
 }
 
+/// A container for a widget yet to be inserted.
+#[non_exhaustive]
+#[expect(missing_docs, reason = "Names are self-explanatory.")]
+pub struct NewWidget<W: ?Sized> {
+    pub widget: Box<W>,
+    pub id: WidgetId,
+    pub options: WidgetOptions,
+    pub properties: Properties,
+}
+
+/// The options a new widget will be created with.
+#[derive(Default, Debug)]
+pub struct WidgetOptions {
+    /// The transform the widget will be created with.
+    pub transform: Affine,
+    /// The disabled state the widget will be created with.
+    pub disabled: bool,
```

Comment:
If doing this, allowing a widget to be created in the stashed state would also be useful. That is probably a follow up issue, but mentioning it here for now.

Reply:
I don't think that's the case, because the stashed state is *only* controlled by the parent.
That being said, there should be a way for a parent to add a widget as stashed.

It's the subtle difference between creating in the stashed state and added in the stashed state; I don't think it makes sense as a field of NewWidget (but I've not carefully studied this)